kyaml/openapi/openapi.go

```diff
@@ -4,15 +4,23 @@
 package openapi
 
 import (
+	"encoding/json"
+	"fmt"
 	"sync"
 
 	"github.com/go-openapi/spec"
+	"sigs.k8s.io/kustomize/kyaml/errors"
 	"sigs.k8s.io/kustomize/kyaml/yaml"
 )
 
-var setup sync.Once
-var schema spec.Schema
-var schemaByResourceType map[yaml.TypeMeta]*spec.Schema
+var c state
+
+type state struct {
```

Comment:
some kind of struct comment please?  or at least a better name

Reply:
done